logistical.js

```diff
@@ -7,23 +7,83 @@
  * Made available via the MIT License, text included in LICENSE.
  */
 
-var math = require('mathjs');
+var Validator = require('./lib/linear_support.js');
 
 /*
  * Main classifier entity.
  */
 var Classifier = function() {};
 
-/* Computes the logistic function value for a given input. */
+/*
+ * Intermediate function for calculating the partial sum for the likelihood
+ *
+ * w = vector coefficients with k elements
+ * Xi = vector of training data with k elements
+ */
+Classifier.prototype.ZiPartialSum = function(w, Xi) {
+  var validator = new Validator();
+
+  validator.isVector(w, Xi);
+
+  var N = validator.hasEqualElementCount(w, Xi);
+
+  var sum = 0;
+
+  /*
+   * @refactor
+   * Shouldn't this just be the dot product of the two vectors?
+   */
+  for ( var k = 1; k <= N; k++) {
+    sum += w.e(k) * Xi.e(k);
+  }
+
+  return sum;
+};
+
+/*
+ * Compute the likelihood for a set of coefficients, data and labels
+ *
+ * w = Vector coefficients
+ * Y = array of labels for the data (Y1, Y2, ..., Yn)
+ * X = Matrix of training data vecors (X1, X2, ..., Xn)
+ */
+Classifier.prototype.likelihood = function(w, Y, X) {
```

Comment:
Technically, this function computes the log likelihood, not the actual likelihood. Should we name it logLikelihood?